java-squid/src/main/java/org/sonar/java/se/ProgramState.java

```diff
@@ -214,4 +236,21 @@ public int constraintsSize() {
   public SymbolicValue getValue(Symbol symbol) {
     return values.get(symbol);
   }
+
+  public void forEachConstraints(PMap.Consumer<SymbolicValue, Object> action) {
+    constraints.forEach(action);
+  }
+
+  public Set<SymbolicValue> getFieldValues() {
+    final HashSet<SymbolicValue> fieldValues = new HashSet<>();
```

Comment:
No special need for final and please use interface. No need to know implementation. I would even recommend to use ImmutableSet from guava for safety.